packages/legend-studio/src/stores/editor-state/element-editor-state/service/ServiceTestState.ts

```diff
@@ -497,39 +513,60 @@ export class SingleExecutionTestState {
   }
 
   *generateTestData(): GeneratorFn<void> {
-    this.isGeneratingTestData = true;
-    // NOTE: here, we attempt to use engine to generate test data.
-    // Once all types of generate data are supported we will move to just using engine
-    let generatedTestData: string | undefined = undefined;
-    const executionInput =
-      this.serviceEditorState.executionState.serviceExecutionParameters;
-    if (executionInput) {
-      try {
-        generatedTestData =
-          (yield this.editorStore.graphManagerState.graphManager.generateMappingTestData(
-            this.editorStore.graphManagerState.graph,
-            executionInput.mapping,
-            executionInput.query,
-            executionInput.runtime,
-            PureClientVersion.VX_X_X,
-          )) as string;
-      } catch (error) {
-        assertErrorThrown(error);
-        this.editorStore.applicationStore.log.error(
-          LogEvent.create(GRAPH_MANAGER_EVENT.EXECUTION_FAILURE),
-          error,
-        );
-      }
-    }
-    if (generatedTestData) {
+    try {
+      this.isGeneratingTestData = true;
+      // NOTE: here, we attempt to use engine to generate test data.
+      // Once all types of generate data are supported we will move to just using engine
+      const executionInput =
+        this.serviceEditorState.executionState.serviceExecutionParameters;
+      assertNonNullable(
+        executionInput,
+        'Service execution context (query, mapping, runtime) is needed to generate test data',
+      );
+      const generatedTestData =
+        (yield this.editorStore.graphManagerState.graphManager.generateMappingTestData(
+          this.editorStore.graphManagerState.graph,
+          executionInput.mapping,
+          executionInput.query,
+          executionInput.runtime,
+          PureClientVersion.VX_X_X,
+          this.buildTestDataParameters(executionInput.query),
+          this.hashGenerateTestData,
+        )) as string;
       singleExecTest_setData(this.test, generatedTestData);
-    } else {
+    } catch (error) {
+      assertErrorThrown(error);
       singleExecTest_setData(this.test, '');
-      this.editorStore.applicationStore.notifyError(
-        `Can't auto-generate test data for service`,
-      );
+      this.editorStore.applicationStore.notifyError(error);
+    } finally {
+      this.isGeneratingTestData = false;
     }
-    this.isGeneratingTestData = false;
+  }
+
+  buildTestDataParameters(rawLambda: RawLambda): (string | number | boolean)[] {
```

Comment:
Let's just make this an utility method instead, exposing through the class is dangerous as it can be mistakenly used by somebody else.